pkg/handlers/v1/books.go

```diff
@@ -2,56 +2,121 @@ package v1
 
 import (
 	"book_service/pkg/clients"
+	_const "book_service/pkg/constants"
 	"book_service/pkg/models/common"
 	"book_service/pkg/models/common/req"
+	"book_service/pkg/models/common/res"
 	"book_service/pkg/query"
 	"book_service/pkg/utils"
-	"fmt"
+	"net/http"
+
 	"github.com/gin-gonic/gin"
 	"github.com/google/uuid"
 	"github.com/jinzhu/copier"
 	"github.com/sirupsen/logrus"
-	"net/http"
 )
 
 var booksIndex, _ = utils.GetEnvVar[string]("BOOKS_INDEX", "books")
 
 func GetBookById(c *gin.Context) {
-	payload, _ := c.Get("validated")
-	bookId, _ := payload.(req.GetBook)
+	bookReq := utils.GetValidatedPayload[req.GetBook](c)
 
-	// TODO: it should be in the validation level with extra validators and transformations
-	_, parseErr := uuid.Parse(bookId.ID)
-	if parseErr != nil {
-		c.JSON(http.StatusBadRequest, gin.H{"error": parseErr})
-		return
-	}
-
-	esQuery := query.NewQueryBuilder().ID(bookId.ID).Build()
-	fmt.Println(esQuery)
-	hits, err := clients.DoSearch(c, booksIndex, esQuery)
+	esQuery := query.NewQueryBuilder().ID(bookReq.ID).Build()
+	hits, _, err := clients.DoSearch(c, booksIndex, esQuery, 1, 0)
 	if err != nil {
-		c.JSON(http.StatusInternalServerError, gin.H{"message": err})
+		logrus.Errorf("Error searching for book with ID %s: %v", bookReq.ID, err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
 		return
 	}
 	if len(hits) == 0 {
+		logrus.Infof("Book with ID %s not found", bookReq.ID)
 		c.JSON(http.StatusNotFound, gin.H{"message": "Book not found"})
 		return
 	}
 
-	c.JSON(200, hits[0])
+	logrus.Infof("Book with ID %s retrieved successfully", bookReq.ID)
+	c.JSON(http.StatusOK, hits[0])
 }
 
 func CreateBook(c *gin.Context) {
-	payload, _ := c.Get("validated")
-	bodyBookReq, _ := payload.(req.AddBook)
-	book := common.Book{ID: uuid.New(), PublishDate: bodyBookReq.PublishDate.Format("2006-01-02")}
+	bodyBookReq := utils.GetValidatedPayload[req.AddBook](c)
+	book := common.Book{
+		ID:          uuid.New(),
+		PublishDate: bodyBookReq.PublishDate.Format("2006-01-02"),
```

Comment:
Look at the constants of `time` package for the date format

Reply:
I need the format `yyyy-mm-dd` but unfortunately IDK why if I specify `PublishDate.Format("yyyy-mm-dd")` is not working so I just putted a random date